src/cljs/org/broadinstitute/firecloud_ui/page/workspace/analysis/igv_utils.cljs

```diff
@@ -0,0 +1,40 @@
+(ns org.broadinstitute.firecloud-ui.page.workspace.analysis.igv-utils
+  (:require
+    [org.broadinstitute.firecloud-ui.common :as common]
+    [org.broadinstitute.firecloud-ui.endpoints :as endpoints]
+    [org.broadinstitute.firecloud-ui.utils :as utils]
+    ))
+
+
+(defn- finder [{:keys [track-url on-success on-error]} options]
+  (let [possible-idxs (atom nil)]
+    (add-watch possible-idxs :loop
+               (fn [_ _ _ new-list]
+                 (if (empty? new-list)
+                   (on-error (str "Index file not found for " track-url))
+                   (let [possible-idx (first new-list)
+                         parsed (common/parse-gcs-uri possible-idx)]
+                     (endpoints/call-ajax-orch
+                       {:endpoint (endpoints/get-gcs-stats (:bucket-name parsed) (:object parsed))
```

Comment:
an option: since you're calling this endpoint anyway, for any given object in GCS you could read the value of "mediaLink" inside the response. The "mediaLink" value should give you the same URL as gcs-object->google-url ... but would supposedly be future-proof, if Google ever changed their url scheme.

No need to update PR for this ... just keep it in mind.